Becoming Mobile First

ashfurrow opened the pull request:
~This is a rough first draft – I'm looking for high-level structural feedback here. Ignore the typos for now please 😅~

Here are some questions I'm looking for feedback on:

- [ ] The early part of the post paints the app situation last year as quite dire, for dramatic effect. Is this too over-the-top?
- [ ] Are there any big pieces I've left out? 
- [ ] Is there anything I'm going into excessive detail about?

Here's what's left to do:

- [x] Address all the todos, which are generally adding images and embeds to break up the text.
- [x] Add lots of links to supporting docs, PRs, etc.
- [x] Add a comments thread.
- [x] Add this to our React Native series.
- [x] Add more categories.
- [x] Proofread (last step, look for typos).

ArtsyOpenSource commented:
<!--
  1 Error: Please add a <!-- more --> tag...
  2 Warnings: Please use React Native with c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add a <!-- more --> tag where you’d like this post to break for post preview</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please use React Native with capitals</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please use fastlane, lowercase ‘f’</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

ashfurrow commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,278 @@
+separated). We overhauled the CI configuration to leverage heavy caching – average build times dropped from fifteen
+minutes to less than five. And we adopted stricter TypeScript compiler settings so that engineers would be forced
+to deal with nullability and other causes of bugs. (Hey, sometimes you need a carrot and sometimes you need a
+stick.)
```
Uhhhh I forgot to mention how we combined Emission and Eigen!

pvinis commented:
Long read, but I think it's great. It feels like it is 3 parts, first is the "old teams and old app", then comes the "overhauled teams and tech work for app" and finally the "current and future teams". I liked the technical parts in the middle, but I don't know if everyone would be interested in that. I feel some details or very top-level things should be there in order to show *how* the change happened, but maybe not too much, maybe it's nicer to keep this post more about the teams and the mobile-first decision and execution? 🤔

After looking again, maybe the "too many details" I mention comes from the sprint/deployment details. That explanation takes a few paragraphs and an image, when it feels it could be a link similar to the iOS Learning Group?

admbtlr commented:
Great post! I like the way you've managed to make it chronological _and_ thematic, so that the meaty topics naturally float up from the story. 

In order to break things up and add some structure, I wonder whether it might be good to signpost those topics a bit more with subheadings - maybe like _Early Wins_, _Spreading the Knowledge_, _Deploy Often_, _DevEx_, _Sharing the Knowledge_, _The Way Forward_ (I'm sure you can come up with much better versions of these!)

A small point: when you start talking about Knowledge Shares, it sounds as if you're mentioning them for the first time, but you actually refer to them a couple of paragraphs earlier. That threw me slightly, and made me backtrack to check whether I had misread the earlier reference.

To answer your questions: I don't think you paint _too_ dire a picture, and it definitely makes the narrative compelling; I think the level of detail is good, but as I said, it might be good to impose some structure (although obviously this is a stylistic preference); and I don't know enough to know whether you've missed anything!

MounirDhahri commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,295 @@
+---
+
+Mike said something that caused a lot of raised eyebrows in the crowd. People were excited. _I_ was excited.
+Artsy's product organization would shift to adopt a "mobile-first strategy."
```
🎉

MounirDhahri commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,295 @@
+Our App Store ranking has shot through the roof – not surprising considering our "make it not suck" and then "make
+it amazing" approach. Artsy's iOS app rating now sits at a stout 4.7.
+
+![Screenshot of a graph showing an increasing average App Store rating](/images/2020-09-29-becoming-mobile-first-at-artsy/ratings.png)
```
this is impressive!

ashfurrow commented:
> That explanation takes a few paragraphs and an image, when it feels it could be a link similar to the iOS Learning Group?

I agree @pvinis, but sadly I don't have a blog post to link to 😢 I wish I'd written one at the time, but we'll have to live with it.

> I wonder whether it might be good to signpost those topics a bit more with subheadings

Thank you @admbtlr – great idea. I've added some H2's, let me know what you think. I took another look at Knowledge Shares, and I agree it's disorienting. I've made sure the first mention alludes to their later discussion, which also alludes back to their earlier mention. Hopefully this helps readers feel more at ease.

---

I've added as many links as I can think of now, it's **final proofreading time**. I'll look this over again tomorrow for typos and would appreciate anyone who can spare some time to do the same 🙇

brainbicycle commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,340 @@
+
+All of this was automated through [fastlane](https://fastlane.tools) on our CI provider. We had been using fastlane
+for a long time at Artsy, but the Mobile Experience team took the time to share knowledge of how it worked. Any
+engineer at Artsy can now make a beta (`make beta`) or promote the latest beta to an App Store submission
```
should this be `make deploy`?  Maybe better to update command than blog haha, `make beta` is pretty nice

ashfurrow replied:
Ah! Yes 😅

brainbicycle commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,340 @@
+support, we now boast a full roster of engineers and product support. I want to thank everyone on the team,
+currently: David, Sam, Brian, Mike, Brittney, Pavlos, and Mounir. You have made the past year an incredibly
+rewarding experience for me as I learn the ropes of technical leadership. I'm so proud of what we've built together
+and I'm excited for what the new Collector Experience team is going to do next!
```
🎉 Awesome read!

admbtlr commented:
> I've added some H2's, let me know what you think. 

Yes, they look great! I think it makes the text much more approachable.

> I took another look at Knowledge Shares, and I agree it's disorienting. I've made sure the first mention alludes to their later discussion, which also alludes back to their earlier mention. Hopefully this helps readers feel more at ease.

Yup, totally 👍

SamRozen commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,340 @@
+
+As the Mobile Experience team formed, I reached out to other mobile teams to find out about how they structured
+regular releases. [Matt Greenwell](https://twitter.com/Cform) from BuzzFeed was really helpful in particular,
+outlining their experience of the pros and cons of a regular release cadence. We implemented a two-week release
```
Really nice to see you mention your process to move to two-week release cadence.

SamRozen commented on _posts/2020-09-29-becoming-mobile-first-at-artsy.markdown:
```diff
@@ -0,0 +1,340 @@
+Like I said, I owe you a whole blog post on Knowledge Share meetings. I hope I have conveyed how important these
+scheduled "structured unstructured learning" times have been for us.
+
+## The Results
```
Nice way to celebrate these big wins! 🎉

SamRozen reviewed (commented):
Great post! Thanks a lot Ash for taking this time to celebrate MX achievements and how we became a better org through this.
